### DISCUSSION TITLE
XLA with ASAN

### DISCUSSION THREAD

**Discussion by apivovarov** [openxla/xla#11812] [General]
Hello Everyone

Do we have any document/article explaining how to build XLA tools with ASAN enabled?

E.g. for the following tool
```bash
bazel build //xla/tools:run_hlo_module

bazel-bin/xla/tools/run_hlo_module \
--input_format=hlo \
--input_module=model.hlotxt \
--platform=CPU
--print_literals=true
```

Thank you
Alex

**Comment by jreiffers** [comment]
Is this still relevant? I assume you've tried `--config=asan`? Can you share the output, if you did?

**Comment by apivovarov** [comment]
ERROR: Config value 'asan' is not defined in any .rc file


I also checked xla/blob/main/.bazelrc
<https://github.com/openxla/xla/blob/main/.bazelrc> file. It does not have
asan config.


tsl .bazelrc <https://github.com/google/tsl/blob/main/.bazelrc>

TF .bazelrc <https://github.com/tensorflow/tensorflow/blob/master/.bazelrc>

do not have asan config too.

**Comment by jreiffers** [comment]
@ddunl This sounds like something you know about :-)

**Comment by ddunl** [comment]
I have not tried this, but I'd be curious to see the results of 
```
build:asan --strip=never
build:asan --copt -fsanitize=address
build:asan --copt -DADDRESS_SANITIZER
build:asan --copt -O1
build:asan --copt -g
build:asan --copt -fno-omit-frame-pointer
build:asan --linkopt -fsanitize=address
```
